test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -15,11 +17,13 @@
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Logging.Testing;
 using Microsoft.Extensions.ObjectPool;
 using Microsoft.Extensions.PlatformAbstractions;
 using Xunit;
 
+[assembly: HostingStartup(typeof(WebHostBuilderTests.TestHostingStartup))]
```

Comment:
Hard to test both 1 and 2, it's a single assembly and I'd rather not add another assembly just to test this (unless we figured that was important somehow).